cf_xarray/accessor.py

```diff
@@ -2335,6 +2376,101 @@ def bounds_to_vertices(
                 )
         return obj
 
+    @property
+    def grid_mappings(self) -> dict[str, list[str]]:
+        """
+        Property that returns a dictionary mapping variable names
+        to the variable names of their grid_mapping.
+
+        Returns
+        -------
+        dict
+            Dictionary mapping variable names to the variable names of their grid_mapping.
+
+        See Also
+        --------
+        Dataset.cf.get_grid_mapping
+        Dataset.cf.get_grid_mapping_name
+
+        References
+        ----------
+        Please refer to the CF conventions document : https://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#grid-mappings-and-projections
+
+        For a list of valid grid_mapping names, refer to: https://cfconventions.org/Data/cf-conventions/cf-conventions-1.10/cf-conventions.html#appendix-grid-mappings
+
+        Examples
+        --------
+        >>> from cf_xarray.datasets import rotds
+        >>> rotds.cf.grid_mappings
+        {'air_temperature': ['rotated_pole'], 'temp': ['rotated_pole']}
```

Comment:
```suggestion
	{'rotated_pole": ['air_temperature', 'temp']}
```

So, our current properties (axes, standard_names, bounds, etc.)have this format : "CF name": "variable name". 

I think we should do that instead. To get the values for a single DataArray, the user can still do `ds.cf["air_temperature"].cf.grid_mapping`

Reply:
Agreed! I didnt' really recognized that logic from the beginning...

Reply:
Should it maybe even be `{'rotated_latitude_longitude": ['air_temperature', 'temp']}`, that's actually the valid `grid_mapping_name` for rotated pole.

Reply:
Ah yes that seems right

Reply:
ok, after all, when if understand correctly, it should work like

* `rotds.cf.grid_mappings` should give: `{'rotated_latitude_longitude": ['rotated_pole']}`
* `rotds.cf["rotated_latitude_longitude"]` should return `rotated_pole` DataArray.

The connection between data variables and grid mappings would work through the DataArray accessors.
Sorry, i was just following the bounds logic in the beginning (which seemed similar)....

Reply:
Is it common to have more than one `grid_mapping` for a Dataset? 

We could also make `DataArray.cf["grid_mapping"] -> DataArray` work and `Dataset.cf["grid_mapping"] -> Dataset` work. 

But we can leave it for later too.

Reply:
It's rather uncommon to have more than one grid_mapping actually! I just felt that it should be possible to access by `grid_mapping_name` (which also follows a little bit the measures logic). I actually like `{'rotated_latitude_longitude": ['air_temperature', 'temp']}` because it gives a good overview!

Reply:
> it should be possible to access by grid_mapping_name

Yup I agree.

It requires you to know the actual grid mapping, which is why I suggested `.cf['grid_mapping']` . But yeah, we can wait for someone to request it.